File: js_base/demos/bases_dom/Person.js
// Création d'une classe dans un autre fichier

class Person {
    constructor(firstname, lastname) {
        this.firstname = firstname;
        this.lastname = lastname;
    }
}

// Autre classe pour l'exemple :
class AutreClasse {
    constructor(titi, toto) {
        this.titi = titi;
        this.toto = toto;
    }
}

// Exportation des classes

// // Exportation de chaque classe :
// export { Person, AutreClasse }

// Exportation par défaut :
export default Person;

File: java_spring/exos/exo6/src/main/java/com/example/exo6/entities/Person.java
package com.example.exo6.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.UUID;

// Annotation JPA pour préciser que les classes enfants
// sont des entités possédant ces attributs en plus des leur,
// et que chaque classe enfant donne une table indépendante
@Entity
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@Data
public abstract class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    protected String email;
    protected String firstName;
    protected String lastName;
    protected LocalDate birthDate;
}
